lib/game.rb

```diff
@@ -0,0 +1,154 @@
+
+class Game
+    attr_reader :player, :frame, :roll, :score, :count_next, :pins_rest , :list_boundaries, :message
+
+    def initialize(player, pins_rest = 10, frame = 1, roll = 0, score = 0, count_next = 0)
+      @player, @frame, @roll, @this_roll_count, @this_roll_bonus = capitalize_name(player), frame, roll, 0, 0
+      @score, @count_next, @pins_rest, @this_roll_messg =  score, count_next, pins_rest, nil
+      @list_boundaries = { pins_rest: 10, frame: 10, roll: 2, score: 300, count_next: 2 }
+      check_initialize_arguments
+    end
+
+    def check_initialize_arguments
+      @list_boundaries.each do |arg, max| 
+        raise 'player has non-number ' << arg.to_s unless eval('@' << arg.to_s).is_a?(Integer) 
+        raise 'player has negative ' << arg.to_s if eval('@' << arg.to_s) < 0 
+        raise 'player has ' << arg.to_s << ' greater than ' << max.to_s if eval('@' << arg.to_s) > max 
+      end
+    end
+
+    def register_pins(pins_number = 0)
```

Comment:
I feel like it doesn't make sense to set a default value for pins. You would not want to register a bowl of 0 unless this is specified.

Reply:
I thought about the situation of fault or miss when we still have to move with the frame.